java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -2349,84 +2349,150 @@ public final ColumnVector stringLocate(Scalar substring, int start, int end) {
    * The number of rows in the output columns will be the same as the input column.
    * Null entries are added for a row where split results have been exhausted.
    * Null string entries return corresponding null output columns.
-   * @param delimiter UTF-8 encoded string identifying the split points in each string.
+   * @param delimiter UTF-8 string identifying the split points or split pattern in each string.
    *                  An empty string indicates split on whitespace.
    * @param maxSplit the maximum number of splits to perform, or -1 for all possible splits.
+   * @param splitByRegex a boolean flag indicating whether the input string will be split by a
+   *                     regular expression pattern or just by a string literal delimiter.
    * @return New table of strings columns.
    */
-  public final Table stringSplit(Scalar delimiter, int maxSplit) {
+  public final Table stringSplit(String delimiter, int maxSplit, boolean splitByRegex) {
```

Comment:
Why make this a breaking change for java when it is not a breaking change for CUDF?  What do we not follow cudf and have an RE version of the method that takes a String and a non-re version that takes the Scalar still?

Reply:
Because the new cudf API (`strings::split_re`) requires a host string scalar input. If we want to call both `strings::split` (requires a `string_scalar` input) and `strings::split_re` (requires `std::string` input) we need to pass a Java String. The JNI function then can construct either an `std::string` or a `string_scalar` from it. Passing a Java `Scalar` cannot do that.

Reply:
One can argue that we can add separate sets of Java `stringSplitRe` and `stringSplitRecordRe` APIs alongside with the existing ones `stringSplit` and `stringSplitRecord`. This is possible but will incur the cost of implementation and maintenance. We won't just add one Java function, instead we will add a set of overload functions. So my solution is to just add a boolean flag to the existing overloads and let the JNI to decide which corresponding cudf API to call.

Reply:
@andygrove Do you have your draft PR using this one up yet?

Reply:
@revans2 ==> https://github.com/NVIDIA/spark-rapids/pull/4714